src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -92,6 +101,19 @@ Status CoreWorkerDirectActorTaskSubmitter::SubscribeActorUpdates() {
     } else {
       // Remove rpc client if it's dead or being reconstructed.
       rpc_clients_.erase(actor_id);
+
+      // For tasks that have been sent and are waiting for replies, treat them
+      // as failed when the destination actor is dead or reconstructing.
+      auto iter = waiting_reply_tasks_.find(actor_id);
```

Comment:
when the actor dies, we should receive an error callback for the `PushTask` request and we'll treat the task as failed in the callback. 
Did you see cases where the callback isn't being called?

Reply:
AFAIK, The callback of `PushTask` will only be invoked if remote returned an error or replied. It wouldn't get executed if the remote crashed.